commit 26a14b65991b79d317638f78dc6b4e5ac1ef712e
Author: Neil Brown <neilb@suse.de>
Date:   Mon Dec 7 17:23:48 2009 -0500

    mount.nfs: Retry v4 mounts with v3 on ENOENT errors
    
    Retry v4 mounts with a v3 mount when the version
    is not explicitly specified and the mount fails
    with ENOENT. The will help deal with Linux servers
    that do not automatically export a pseudo root
    
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/mount/stropts.c b/utils/mount/stropts.c
index a0b9e7f..4007150 100644
--- a/utils/mount/stropts.c
+++ b/utils/mount/stropts.c
@@ -668,9 +668,10 @@ static int nfs_try_mount(struct nfsmount_info *mi)
 				/* 
 				 * To deal with legacy Linux servers that don't
 				 * automatically export a pseudo root, retry
-				 * ENOENT errors using version 3
+				 * ENOENT errors using version 3. And for
+				 * Linux servers prior to 2.6.25, retry EPERM
 				 */
-				if (errno != ENOENT)
+				if (errno != ENOENT && errno != EPERM)
 					break;
 			}
 		}
